packages/financial-templates-lib/src/price-feed/ForexDailyPriceFeed.js

```diff
@@ -230,13 +233,21 @@ class ForexDailyPriceFeed extends PriceFeedInterface {
       .subtract(16, "hours")
       .format("YYYY-MM-DD");
   }
-  _dateTimeToSecond(inputDateTime) {
+  _dateTimeToSecond(inputDateTime, endOfDay = false) {
     // To convert from date string to unix, we assume that the date string
-    // denotes CET time, and then we add 16 hours since the ECB "begins" days at 16:00 CET..
-    return moment
-      .tz(inputDateTime, "YYYY-MM-DD", "Europe/Berlin")
-      .add(16, "hours")
-      .unix();
+    // denotes CET time, and then we add 16 hours since the ECB "begins" days at 16:00 CET.
+    if (endOfDay) {
+      return moment
+        .tz(inputDateTime, "YYYY-MM-DD", "Europe/Berlin")
+        .endOf("day")
+        .add(16, "hours")
+        .unix();
+    } else {
+      return moment
+        .tz(inputDateTime, "YYYY-MM-DD", "Europe/Berlin")
+        .add(16, "hours")
+        .unix();
+    }
```

Comment:
What is this change?

Reply:
So before I added the `.endOf("day")`, the conversion from datetime string to unix was using the BEGINNING of the day by default, but sometimes I actually wanted the end of the day. For example, I was setting closeTime = beginning of the day (unintnetionally), and subtrackign 24 hours to get the openTime, which was actually setting the time to 24 hours behind what it should be.